### PR TITLE
[Terrain] Bulk query unit tests

### PR DESCRIPTION
**Pull request by mbalfour-amzn**
This adds unit tests to verify that all of our terrain query APIs produce equivalent results. In creating this tests, I also changed the following:
* Fixed a bug where the bulk Normal queries were _not_ always returning the same results as the single-value queries.
* Fixed a bug where the Async cancel unit test wasn't setting the number of jobs to 1, even though it relied on that.
* Cleaned up all the unit tests to use GemTestEnvironment so that we can use gradient / shape / surface components safely in our tests and benchmarks. This also cleaned up a lot of redundant setup / teardown code.
* Changed the benchmarks to use a real shape instead of a mock shape so that the benchmarks are more accurate.
* Removed the unused Terrain Editor test files.

### CODE REVIEW

**Review by amzn-sj — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp [hunk 1]
```diff
@@ -0,0 +1,938 @@
...
+            // Verify that every value is found exactly once in each set. The two sets might not have the values in the same order though,
+            // so we need to search for each value, verify it's found, and verify that it hadn't previously been found.
+            AZStd::vector<bool> matchFound(baselineValues.size(), false);
+            for (auto& comparisonValue : comparisonValues)
+            {
+                auto foundValue = AZStd::find(baselineValues.begin(), baselineValues.end(), comparisonValue);
+                EXPECT_NE(foundValue, baselineValues.end());
+                if (foundValue != baselineValues.end())
+                {
+                    EXPECT_FALSE(matchFound[foundValue - baselineValues.begin()]);
```
Comment (invertednormal) [inline Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp:73]: nit: consider making `foundValue - baselineValues.begin()` a separate variable like `valueIndex` to make this a little more clear.

Comment (invertednormal) [reply Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp:73]: looks like you actually did this in at least one place below 😛 best to make it consistent.

**Review by invertednormal — APPROVED** [review]
looks good, just a few nits / suggestions.

File: Gems/Terrain/Code/Tests/TerrainTestFixtures.cpp [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        // Create a Gradient Transform Component with arbitrary parameters.
+        GradientSignal::GradientTransformConfig gradientTransformConfig;
+        gradientTransformConfig.m_wrappingType = GradientSignal::WrappingType::None;
+        entity->CreateComponent<GradientSignal::GradientTransformComponent>(gradientTransformConfig);
+
+        ActivateEntity(entity.get());
+        return entity;
+    }
+
+        AZStd::unique_ptr<AZ::Entity> TerrainBaseFixture::CreateTestLayerSpawnerEntity(
```
[the remark this answers is not in the document]
Comment (mbalfour-amzn) [reply Gems/Terrain/Code/Tests/TerrainTestFixtures.cpp]: Fixed!

File: Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp [hunk 3]
```diff
@@ -0,0 +1,938 @@
...
+            AzFramework::Terrain::TerrainDataRequests::Sampler sampler,
+            AZStd::vector<AZ::Vector3>& queryPositions,
+            AZStd::vector<AZ::Vector3>& resultPositions)
+        {
+            queryPositions.clear();
+            resultPositions.clear();
+
+            auto perPositionCallback = [&queryPositions, &resultPositions](
+                                              [[maybe_unused]] size_t xIndex, [[maybe_unused]] size_t yIndex,
+                                              const AzFramework::SurfaceData::SurfacePoint& surfacePoint, bool terrainExists)
```
[the remark this answers is not in the document]
Comment (mbalfour-amzn) [reply Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp]: Fixed!

File: Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp (hunk 1 above)
Comment (mbalfour-amzn) [reply Gems/Terrain/Code/Tests/TerrainBulkQueryTests.cpp]: Fixed!

**Pull request merged**